### PR TITLE
[openuv] Correcting NPE in IconProvider

### LINKED ISSUE #15464: [openuv] NPE generated by the iconserver
I observed this in the log when browsing items in UI : 

```
2023-08-20 02:55:15.081 [WARN ] [org.eclipse.jetty.server.HttpChannel] - /icon/uv-index
java.lang.NullPointerException: Cannot invoke "java.net.URL.openStream()" because "iconResource" is null
	at org.openhab.binding.openuv.internal.OpenUVIconProvider.getResource(OpenUVIconProvider.java:118) ~[?:?]
	at org.openhab.binding.openuv.internal.OpenUVIconProvider.getIcon(OpenUVIconProvider.java:100) ~[?:?]
	at org.openhab.core.ui.icon.internal.IconServlet.doGet(IconServlet.java:143) ~[?:?]
```
This is generated if UV state is 0.0

### PR DESCRIPTION
**Pull request by clinique**
Resolves #15464.

Avoids loading an icon that does not exist.
Adds UV 0 icon.

### CODE REVIEW

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java [hunk 1]
```diff
@@ -115,10 +115,12 @@ private String getResource(String iconName) {
...
         URL iconResource = context.getBundle().getEntry("icon/%s.svg".formatted(iconName));
-        try (InputStream stream = iconResource.openStream()) {
-            result = new String(stream.readAllBytes(), StandardCharsets.UTF_8);
-        } catch (IOException e) {
-            logger.warn("Unable to load ressource '{}' : {}", iconResource.getPath(), e.getMessage());
+        if (iconResource != null) {
+            try (InputStream stream = iconResource.openStream()) {
+                result = new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                logger.warn("Unable to load ressource '{}' : {}", iconResource.getPath(), e.getMessage());
```
Comment (jlaur) [inline bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java:122]:
```suggestion
                logger.warn("Unable to load resource '{}': {}", iconResource.getPath(), e.getMessage());
```

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java [hunk 2]
```diff
@@ -115,10 +115,12 @@ private String getResource(String iconName) {
         String result = "";
 
         URL iconResource = context.getBundle().getEntry("icon/%s.svg".formatted(iconName));
-        try (InputStream stream = iconResource.openStream()) {
-            result = new String(stream.readAllBytes(), StandardCharsets.UTF_8);
-        } catch (IOException e) {
-            logger.warn("Unable to load ressource '{}' : {}", iconResource.getPath(), e.getMessage());
+        if (iconResource != null) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java:118]: Should anything be logged if we didn't find an entry here (like we log when unable to load the resource)? For example, the `iconName` might be relevant to know?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the fix! Added minor comments regarding logging.

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java (hunk 1 above)
Comment (clinique) [reply bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java:122]: Changed

File: bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java (hunk 2 above)
Comment (clinique) [reply bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/OpenUVIconProvider.java:118]: You're more than correct. Let me correct this.

**Review by jlaur — APPROVED** [review]
Thanks for the fix!

**Comment by jlaur** [comment]
@clinique - there's a problem with you sign-off not carrying your full name. Can you have a look? I have corrected it for the squash commit for this PR.

**Pull request merged**